src/App.jsx

```diff
@@ -14,45 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goodVisible, setGoodVisible] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
+
+  const sortAlphabet = () => {
+    if (sortField.includes('reverse')) {
+      setGoodVisible([...goodVisible].sort((a, b) => b.localeCompare(a)));
+      setSortField('alphabet reverse');
+    } else {
+      setGoodVisible([...goodVisible].sort((a, b) => a.localeCompare(b)));
+      setSortField('alphabet');
+    }
+  };
+
+  const sortLength = () => {
+    if (sortField.includes('reverse')) {
+      setGoodVisible([...goodVisible].sort((a, b) => b.length - a.length));
+      setSortField('length reverse');
+    } else {
+      setGoodVisible([...goodVisible].sort((a, b) => a.length - b.length));
+      setSortField('length');
+    }
+  };
+
+  const reverse = () => {
+    if (sortField.includes('reverse')) {
+      setGoodVisible([...goodVisible].reverse());
+      setSortField(sortField.split(' ')[0]);
+    } else {
+      setGoodVisible([...goodVisible].reverse());
+      setSortField(`${sortField} reverse`);
+    }
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            sortField.includes('alphabet')
```

Comment:
Avoid using string concatenation for defining classNames. This can cause unexpected issues. Consider using a library for classNames or an array join.